src/dwalk/dwalk.c

```diff
@@ -417,15 +541,97 @@ int main(int argc, char** argv)
+            case 'c':
+                tr = mfu_pred_relative(optarg, now_t);
+                mfu_pred_add(pred_head, MFU_PRED_CMIN, (void *)tr);
+                break;
+
+            case 'A':
+                tr = mfu_pred_relative(optarg, now_t);
+                mfu_pred_add(pred_head, MFU_PRED_ATIME, (void *)tr);
+                break;
+            case 'M':
+                tr = mfu_pred_relative(optarg, now_t);
+                mfu_pred_add(pred_head, MFU_PRED_MTIME, (void *)tr);
+                break;
+            case 'C':
+                tr = mfu_pred_relative(optarg, now_t);
+                mfu_pred_add(pred_head, MFU_PRED_CTIME, (void *)tr);
+                break;
+
+            case 'B':
+                    t = get_mtimes(optarg);
+                    if (t == NULL) {
+                        if (rank == 0) {
+                            printf("%s: can't find file %s\n", argv[0], optarg);
+                        }
+                   exit(1);
+                }
+                mfu_pred_add(pred_head, MFU_PRED_ANEWER, (void *)t);
+                break;
+            case 'D':
+                t = get_mtimes(optarg);
+                if (t == NULL) {
+                    if (rank == 0) {
+                       printf("%s: can't find file %s\n", argv[0], optarg);
+                    }
+                exit(1);
+                }
+                mfu_pred_add(pred_head, MFU_PRED_CNEWER, (void *)t);
+                break;
+
+            case 'g':
+                /* TODO: error check argument */
+                buf = MFU_STRDUP(optarg);
+                mfu_pred_add(pred_head, MFU_PRED_GID, (void *)buf);
+                break;
+        
+            case 'G':
+                buf = MFU_STRDUP(optarg);
+                mfu_pred_add(pred_head, MFU_PRED_GROUP, (void *)buf);
+                break;
+        
+            case 'u':
+                /* TODO: error check argument */
+                buf = MFU_STRDUP(optarg);
+                mfu_pred_add(pred_head, MFU_PRED_UID, (void *)buf);
+                break;
+        
+            case 'U':
+                buf = MFU_STRDUP(optarg);
+                mfu_pred_add(pred_head, MFU_PRED_USER, (void *)buf);
+                break;
+
+            case 'S':
+                buf = MFU_STRDUP(optarg);
+                mfu_pred_add(pred_head, MFU_PRED_SIZE, (void *)buf);
+                break;
+
+            case 'T':
+                ret = add_type(pred_head, *optarg);
+                if (ret != 1) {
+                    if (rank == 0) {
+                    printf("%s: unsupported file type %s\n", argv[0], optarg);
+                    }
+                exit(1);
+                }
+                break;
+        
             case 'v':
                 mfu_debug_level = MFU_LOG_VERBOSE;
                 break;
             case 'q':
                 mfu_debug_level = 0;
                 break;
-            case 't':
-                text = 1;
-                break;
```

Comment:
Let's keep the ``--text`` option active for a bit, so we can ease people over in to switching.